File: examples/python/basic.py
"""Python example: store, search, and page through memories.

Run a local server first:  npx remembra --http
Then:                      python3 examples/python/basic.py
"""

from __future__ import annotations

import os

from remembra import Remembra

client = Remembra(
    os.environ.get("REMEMBRA_ENDPOINT", "http://127.0.0.1:8787"),
    api_key=os.environ.get("REMEMBRA_API_KEY"),
)

stored = client.store(
    {
        "type": "decision",
        "content": "The payments cluster is upgraded during the tuesday 02:00 UTC window.",
        "scope": "global",
        "tags": ["deploy", "payments"],
        "importance": 8,
    }
)
print("stored", stored["id"])

found = client.search({"query": "payments cluster upgrade window", "limit": 5})
for hit in found["results"]:
    print(hit["memory"]["id"], hit["memory"]["content"])

# Cursor pagination: follow the server's opaque cursor until it stops.
for page in client.iter_list({"limit": 20}):
    print("page:", len(page["memories"]), "active memories")

File: examples/node/basic.mjs
/**
 * Node.js example: store, search, and reuse a memory.
 *
 * Run a local server first:  npx remembra --http
 * Then:                      node examples/node/basic.mjs
 */
import { Remembra } from "@hilbras/remembra/sdk";

const client = new Remembra({
  endpoint: process.env.REMEMBRA_ENDPOINT ?? "http://127.0.0.1:8787",
  apiKey: process.env.REMEMBRA_API_KEY,
});

const stored = await client.store({
  type: "decision",
  content: "Payments deploys happen on tuesday 02:00 UTC with rolling restarts.",
  scope: "global",
  tags: ["deploy", "payments"],
  importance: 8,
});
console.log("stored", stored.id);

// A later session recalls it with the same search the agent would use.
const found = await client.search({ query: "when do payments deploys happen", limit: 5 });
for (const hit of found.results) {
  console.log(`${hit.memory.id}  ${hit.memory.content}`);
}

// Safe retries apply to reads only; a mutation is never retried automatically.
const page = await client.list({ limit: 20, includeArchived: false });
console.log(`active memories: ${page.memories.length}`);
